concepts/functions/about.md

```diff
@@ -0,0 +1,153 @@
+# About
+
+A function allows to group code into a reusable unit.
+There are multiple ways to define functions in JavaScript.
+Here we will look at function declarations and function expressions.
+Other possibilities like [arrow functions][concept-arrow-functions] will be covered in other concepts.
+
+## Function Declaration
+
+The standard way of defining a function in JavaScript is a function declaration, also called function definition or function statement.
+
+It consists of the `function` keyword, the name of the function and a comma-separated list of parameters in round brackets.
+This is followed by the function body (the code that should be executed) wrapped in curly brackets.
+
+```javascript
+function someName(param1, param2, param3) {
+  // ...
+}
+```
+
+## Using a Function
+
+In JavaScript a function is invoked (called) by stating the function name followed by round brackets that contain the arguments.
+
+```javascript
+someName(arg1, arg2, arg3);
+```
+
+Just stating the function name does **not** call the function in JavaScript.
+
+```javascript
+function sayHello() {
+  console.log('Hello, World!');
+}
+
+sayHello;
+// => [Function: sayHello]
+
+sayHello();
+// => 'Hello, World!'
+```
+
+## Parameters
+
+When working with parameters inside the function body, be aware of possible effects to the original value that was passed to the function.
+In JavaScript, the behavior depends on the data type of the argument.
+
+- All values that have primitive data types ([full list here][mdn-primitives]) are immutable in JavaScript, so if used as arguments, they are _passed by value_.
+  That means you are dealing with a copy of the original value in the function body and you can modify it without affecting the original value.
+- All other values (objects, arrays, functions) are _passed by reference_.
+  If you modify arguments of non-primitive types, you are changing the original value outside of the function because the argument represents a reference to the original value, not a copy of that value.
+
+By default, all parameters defined in the function declaration are optional in JavaScript.
+If you provide less arguments than there are parameters, the missing arguments will be `undefined` inside the function, see [Null and Undefined][concept-null-undefined].
+In many cases it makes sense to assign a more appropriate default value than `undefined`. This can by done by specifying default parameters directly in the function definition.
```

Comment:
```suggestion
In many cases it makes sense to assign a more appropriate default value than `undefined`.
This can by done by specifying default parameters directly in the function definition.
```

Reply:
fixed